ner_v2/detectors/temporal/time/en/time_detection.py

```diff
@@ -222,39 +267,54 @@ def _detect_range_12_hour_format(self, time_list=None, original_list=None):
             time_list = []
         if original_list is None:
             original_list = []
-        patterns = re.findall(
-            r'\s((0?[2-9]|0?1[0-2]?)[\s-]*(?::|\.|\s)?[\s-]*?([0-5][0-9])[\s-]*?(pm|am|a\.m|p\.m)[\s-]*?to[\s-]'
-            r'*?(0?[2-9]|0?1[0-2]?)[\s-]*(?::|\.|\s)?[\s-]*?([0-5][0-9])[\s-]*?(pm|am|a\.m|p\.m))',
-            self.processed_text.lower())
+        regex_patterns = re.compile(
+            r'\b(({timezone})?\s*(0?[2-9]|0?1[0-2]?)[\s-]*(?::|\.|\s)?[\s-]*?([0-5][0-9])[\s-]*?(pm|am|a\.m|p\.m)'
+            r'[\s-]*?({timezone})?\s*to[\s-]*?({timezone})?\s*(0?[2-9]|0?1[0-2]?)[\s-]*'
+            r'(?::|\.|\s)?[\s-]*?([0-5][0-9])[\s-]*?(pm|am|a\.m|p\.m)\s*({timezone})?)\b'
+            .format(timezone=self.timezone_choices)
+        )
+        patterns = regex_patterns.findall(self.processed_text.lower())
         for pattern in patterns:
-            original1 = pattern[0]
-            original2 = pattern[0]
+            original1 = pattern[0].strip()
+            original2 = pattern[0].strip()
             if self.departure_flag:
                 time_type = 'departure'
             elif self.return_flag:
                 time_type = 'return'
             else:
                 time_type = None
-            t1 = pattern[1]
-            t2 = pattern[2]
-            ap1 = pattern[3]
+            t1 = pattern[2]
+            t2 = pattern[3]
+            ap1 = pattern[4]
+            tz1 = pattern[1]
+            tz2 = pattern[5]
+            tz = None
+            if tz1 or tz2:
+                tz = self.convert_to_pytz_format(tz1 or tz2)
             time1 = {
                 'hh': int(t1),
                 'mm': int(t2),
                 'nn': str(ap1).lower().strip('.'),
+                'tz': tz or self.timezone.zone,
                 'range': 'start',
                 'time_type': time_type
             }
             time1['nn'] = 'am' if 'a' in time1['nn'] else time1['nn']
             time1['nn'] = 'pm' if 'p' in time1['nn'] else time1['nn']
 
-            t3 = pattern[4]
-            t4 = pattern[5]
-            ap2 = pattern[6]
+            t3 = pattern[7]
+            t4 = pattern[8]
+            ap2 = pattern[9]
+            tz3 = pattern[6]
+            tz4 = pattern[10]
+            tz = None
+            if tz3 or tz4:
+                tz = self.convert_to_pytz_format(tz3 or tz4)
             time2 = {
                 'hh': int(t3),
                 'mm': int(t4),
                 'nn': str(ap2).lower().strip('.'),
+                'tz': tz or self.timezone.zone,
```

Comment:
Since in init, default is UTC, is it returning timezone when no timezone in either message or api?

Reply:
yes, It'll return UTC as default.